ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/TPraos.hs

```diff
@@ -355,47 +360,51 @@ instance SL.PraosCrypto c => ConsensusProtocol (TPraos c) where
 
       SL.GenDelegs dlgMap = SL.lvGenDelegs lv
 
-  tickChainDepState cfg@TPraosConfig{..}
-                    (TickedPraosLedgerView lv)
-                    slot
-                    (TPraosState lastSlot st) =
+  projectHorizonView _cfg _lv = TickedTrivial
+
+  -- TODO comment explaining why we're not using the 'SL.tickChainDepState' function
```

Comment:
Also TODO: open a PR upstream deleting that function?

Reply:
I opened #350 for the natural larger task.

We could also go the 'opposite" route and add a property test that our vendored tick function behaves just as the one from ledger (for an interim period until #350 is done).
I locally tested that this works (and that introducing bugs in TPraos `tickChainDepState` cause the test to fail), but I am not sure if it is worth it:
<details>
<summary>Click to see compatibility test for TPraos ticking</summary>

```haskell
{-# LANGUAGE FlexibleInstances  #-}
{-# LANGUAGE NamedFieldPuns     #-}
{-# LANGUAGE StandaloneDeriving #-}

module Test.Consensus.Protocol.TPraos.LedgerCompat (tests) where

import qualified Cardano.Ledger.BaseTypes as SL
import qualified Cardano.Ledger.Chain as SL
import           Cardano.Ledger.Crypto
import qualified Cardano.Protocol.TPraos.API as SL
import           Cardano.Slotting.EpochInfo.API (epochInfoEpoch,
                     generalizeEpochInfo)
import           Data.Functor.Identity
import           Ouroboros.Consensus.Block
import           Ouroboros.Consensus.HardFork.History.EpochInfo
                     (toPureEpochInfo)
import           Ouroboros.Consensus.HardFork.History.Util (addSlots)
import           Ouroboros.Consensus.Protocol.Abstract
import           Ouroboros.Consensus.Protocol.TPraos
import           Test.Cardano.Ledger.Shelley.Utils (testGlobals)
import           Test.Cardano.Protocol.TPraos.Arbitrary ()
import           Test.QuickCheck
import           Test.Tasty
import           Test.Tasty.QuickCheck

tests :: TestTree
tests = testGroup "TPraos Ledger compat"
    [ testProperty "tick" testTickCompat
    ]

data TickSetup p = TickSetup {
    tsConsensusConfig  :: Blind (ConsensusConfig p)
  , tsTickedLedgerView :: Ticked (LedgerView p)
  , tsSlotNo           :: SlotNo
  , tsChainDepState    :: ChainDepState p
  }

instance Crypto c => Arbitrary (TickSetup (TPraos c)) where
  arbitrary = do
      initialNonce <- arbitrary
      let tpraosParams  = TPraosParams {
              tpraosSlotsPerKESPeriod = SL.slotsPerKESPeriod                 globals
            , tpraosLeaderF           = SL.activeSlotCoeff                   globals
            , tpraosSecurityParam     = SecurityParam $ SL.securityParameter globals
            , tpraosMaxKESEvo         = SL.maxKESEvo                         globals
            , tpraosQuorum            = SL.quorum                            globals
            , tpraosMaxMajorPV        = MaxMajorProtVer $ SL.maxMajorPV      globals
            , tpraosMaxLovelaceSupply = SL.maxLovelaceSupply                 globals
            , tpraosNetworkId         = SL.networkId                         globals
            , tpraosInitialNonce      = initialNonce
            , tpraosSystemStart       = SL.systemStart                       globals
            }
          tsConsensusConfig = Blind TPraosConfig {
              tpraosParams
            , tpraosEpochInfo = generalizeEpochInfo $ SL.epochInfoPure globals
            }
      chainCheckPParams   <-
        SL.ChainChecksPParams <$> arbitrary <*> arbitrary <*> arbitrary
      tsTickedLedgerView  <- fmap TickedPraosLedgerView $
        SL.LedgerView <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> pure chainCheckPParams
      slotAlreadyTickedTo <- frequency
        [ (9 , NotOrigin <$> chooseEnum (0, 1000))
        , (1, pure Origin)
        ]
      slotOffset          <- chooseEnum (0, 200)
      chainDepState       <- SL.ChainDepState <$> arbitrary <*> arbitrary <*> arbitrary
      let tsChainDepState = TPraosState {
              tpraosStateLastSlot      = slotAlreadyTickedTo
            , tpraosStateChainDepState = chainDepState
            }
          tsSlotNo = slotOffset `addSlots` fromWithOrigin 0 slotAlreadyTickedTo
      pure TickSetup {
          tsConsensusConfig
        , tsTickedLedgerView
        , tsSlotNo
        , tsChainDepState
        }
    where
      globals :: SL.Globals
      globals = testGlobals

testTickCompat :: Blind (TickSetup (TPraos StandardCrypto)) -> Property
testTickCompat (Blind ts) =
    consensusImpl === ledgerImpl
  where
    consensusImpl =
          tickedTPraosStateChainDepState
        $ tickChainDepState cfg tlv tsSlotNo cds

    ledgerImpl =
        SL.tickChainDepState
          (mkShelleyGlobals cfg)
          (getTickedPraosLedgerView tlv)
          tickingToNewEpoch
          (tpraosStateChainDepState cds)

    tickingToNewEpoch =
        epochNo (NotOrigin tsSlotNo) /= epochNo (tpraosStateLastSlot cds)
      where
        ei      = toPureEpochInfo $ tpraosEpochInfo cfg
        epochNo = runIdentity . epochInfoEpoch ei . fromWithOrigin 0

    TickSetup {
        tsConsensusConfig  = Blind cfg
      , tsTickedLedgerView = tlv
      , tsSlotNo
      , tsChainDepState    = cds
      } = ts
```

</details>